src/solver/lu_decomposition.cu

```diff
@@ -62,6 +62,10 @@ namespace micm
               size_t A_idx = d_aik[aik_offset++] + tid;
               d_U[U_idx] = d_A[A_idx];
             }
+            else {
+              size_t U_idx = d_uik_nkj[uik_nkj_offset].first + tid;
```

Comment:
`Line 61` has the same definition. Maybe we could define it outside if conditional?

Reply:
I was noticing that too. There's a lot of redefinition in these and I was wondering if now would be a good time to remove those

Reply:
Yeah I think we can address them in this PR.